homeassistant/components/thread/dataset_store.py

```diff
@@ -310,6 +325,62 @@ def preferred_dataset(self, dataset_id: str) -> None:
         self._preferred_dataset = dataset_id
         self.async_schedule_save()
 
+    async def _set_preferred_dataset_if_only_network(
+        self, dataset_id: str, border_agent_id: str
+    ) -> None:
+        """Set the preferred dataset, unless there are other routers present."""
+        _LOGGER.debug(
+            "_set_preferred_dataset_if_only_network called for router %s",
+            border_agent_id,
+        )
+
+        own_router_evt = Event()
+        other_router_evt = Event()
+
+        @callback
+        def router_discovered(
+            key: str, data: discovery.ThreadRouterDiscoveryData
+        ) -> None:
+            """Handle router discovered."""
+            _LOGGER.debug("discovered router with id %s", data.border_agent_id)
+            if data.border_agent_id and data.border_agent_id == border_agent_id:
```

Comment:
```suggestion
            if data.border_agent_id == border_agent_id:
```

Reply:
Right, None check is not necessary. Thanks! :+1: